### PR TITLE
Add `sevctl measurement build`

### PR DESCRIPTION
**Pull request by crobinso**
This adds `sevctl measurement build` for spitting out a measurement+nonce blob, like we get from qemu and libvirt for a running VM.

Patches tell the story. One note is that this doesn't use `sev` crate existing Session bits to build the measurement. The `sev` crate will need some tweaks for us to use it here. And in the end it would only save maybe 7-8 lines of code.

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -329,11 +333,16 @@ fn ca_chain_builtin(chain: &sev::Chain) -> Result<ca::Chain> {
...
+    env_logger::init();
+
     let sevctl = Sevctl::from_args();
     let status = match sevctl.cmd {
         SevctlCmd::Export { full, destination } => export::cmd(full, destination),
         SevctlCmd::Generate { cert, key } => generate::cmd(cert, key),
         SevctlCmd::Ok { gen } => ok::cmd(gen, sevctl.quiet),
+        SevctlCmd::Measurement(option) => match option {
+            measurement::MeasurementCmd::Build(args) => measurement::build_cmd(args),
+        },
```
Comment (tylerfanelli) [inline src/main.rs:367]: Is this match here in anticipation in more `sevctl measurement` commands in the future?

**Review by tylerfanelli — APPROVED** [review]
LGTM.

File: src/main.rs (hunk 1 above)
Comment (crobinso) [reply src/main.rs:367]: I don't have any in mind. I was following the `vmsa` subcommand examples.

**Comment by crobinso** [comment]
@tylerfanelli I added `secret build` commits on top of this PR, since they share some code. Please review + approve when you get some time, thanks!

File: src/main.rs [hunk 2]
```diff
@@ -347,14 +401,11 @@ fn main() {
         },
     };
 
-    if let Err(err) = status {
-        if sevctl.quiet {
-            exit(1);
-        }
-        eprintln!("error: {}", err);
-
+    if status.is_err() && sevctl.quiet {
```
Comment (tylerfanelli) [inline src/main.rs:404]: In this case, wouldn't an error message never be printed?

File: src/measurement.rs [hunk 3]
```diff
@@ -0,0 +1,304 @@
...
+    if let Some(nonce_arg) = args.nonce {
+        nonce = parse_base64_or_path("--nonce", &nonce_arg)?;
+    } else if let Some(launch_measure_blob_arg) = args.launch_measure_blob {
+        let blob = parse_base64_or_path("--launch-measure-blob", &launch_measure_blob_arg)?;
+        nonce = blob[32..].to_vec();
+    } else {
+        return Err(anyhow::anyhow!(
+            "One of --nonce or --launch-measure-blob must be specified."
+        ));
+    }
```
Comment (tylerfanelli) [inline src/measurement.rs:271]: Just a suggestion.. if you don't prefer this change then it's no big deal.

```suggestion
    let nonce = if let Some(nonce_arg) = args.nonce {
        parse_base64_or_path("--nonce", &nonce_arg)?;
    } else if let Some(launch_measure_blob_arg) = args.launch_measure_blob {
        let blob = parse_base64_or_path("--launch-measure-blob", &launch_measure_blob_arg)?;
        blob[32..].to_vec();
    } else {
        return Err(anyhow::anyhow!(
            "One of --nonce or --launch-measure-blob must be specified."
        ));
    }
```

**Review by tylerfanelli — DISMISSED** [review]
Thanks Cole. I've just left 2 comments, however they are not immediate problems. Let me know what you think, and we can merge.

File: src/main.rs (hunk 2 above)
Comment (crobinso) [reply src/main.rs:404]: Yes, but that's how it's expected to work with the `--quiet` flag. See where old code just called `exit(1)`

File: src/measurement.rs (hunk 3 above)
Comment (crobinso) [reply src/measurement.rs:271]: Hmm. I used github's commit suggestion here but didn't try testing locally, and then CI is failing. Trying to resolve it locally is non-obvious to me, so I dropped the patch and repushed. Maybe handle it as a separate PR when this hits git?

File: src/main.rs (hunk 2 above)
Comment (tylerfanelli) [reply src/main.rs:404]: Right, but there is no case for `status.is_err() && !sevctl.quiet`, where we could assume that a user would like something printed, no?

File: src/measurement.rs (hunk 3 above)
Comment (tylerfanelli) [reply src/measurement.rs:271]: Sure, that's fine.

File: src/main.rs (hunk 2 above)
Comment (crobinso) [reply src/main.rs:404]: Sorry, missed this follow up. If you check the full diff, we exit the process when this condition is true. So `status.is_err() && !sevctl.quiet` is implicitly handled by the plain function return value, which prints proper error output and sets proper error code.

**Review by tylerfanelli — APPROVED** [review]

**Pull request merged**